### PR TITLE
Fix provisiong scale group and elb logic

### PR DESCRIPTION
**Pull request by michaelgugino**
Some changes necessary to support provisioning
and scale group logic

Some of this work is duplicated from an existing PR.  That PR is about to merge, so I'll rebase once that hits.

### CODE REVIEW

File: roles/openshift_aws/tasks/provision.yml [hunk 1]
```diff
@@ -34,14 +34,14 @@
   include: elb.yml
   vars:
     openshift_aws_elb_direction: internal
-    openshift_aws_elb_name: "{{ openshift_aws_clusterid }}-{{openshift_aws_node_group_type }}-internal"
+    l_openshift_aws_elb_name: "{{ openshift_aws_elb_name }}-internal"
```
Comment (kwoodson) [inline roles/openshift_aws/tasks/provision.yml]: Why the change here?

File: roles/openshift_aws/tasks/elb.yml [hunk 2]
```diff
@@ -29,9 +29,9 @@
                    if 'master' in openshift_aws_node_group_type or 'infra' in openshift_aws_node_group_type
                    else openshift_aws_elb_listeners }}"
 
-- name: "Create ELB {{ openshift_aws_elb_name }}"
+- name: "Create ELB {{ l_openshift_aws_elb_name }}"
   ec2_elb_lb:
-    name: "{{ openshift_aws_elb_name }}"
+    name: "{{ l_openshift_aws_elb_name }}"
```
Comment (kwoodson) [inline roles/openshift_aws/tasks/elb.yml]: Why this change here?

I think this should be `openshift_aws_elb_name`.  The `elb.yml` can be called with `tasks_from` and would require the user to create `l_openshift_aws_elb_name`.

File: roles/openshift_aws/tasks/provision.yml (hunk 1 above)
Comment (michaelgugino) [reply roles/openshift_aws/tasks/provision.yml]: This is overriding the inventory variable openshift_aws_elb_name.

If this were to be set with extra_vars, it would force-drop -internal and -external.

If someone wants to create and call elb directly, they could set l_openshift_aws_elb_name in inventory.

Comment (kwoodson) [reply roles/openshift_aws/tasks/provision.yml]: I see what you mean. We should probably test to see if it's set and take that. If not, continue using the logic for internal and external. I'd prefer that if someone calls elb directly they use the openshift variable rather than the l_ version.

Comment (michaelgugino) [reply roles/openshift_aws/tasks/provision.yml]: I agree.  That would be good enhancement.  For me, this is a bug fix until we can sanitize clusterid.

**Comment by michaelgugino** [comment]
This branch looks to be in good shape, git was able to merge changes in master automatically, no rebase need at this time.

File: roles/openshift_aws/tasks/launch_config.yml [hunk 3]
```diff
@@ -4,13 +4,18 @@
...
+
 - name: fetch the security groups for launch config
   ec2_group_facts:
     filters:
-      group-name:
-      - "{{ openshift_aws_clusterid }}"  # default sg
-      - "{{ openshift_aws_clusterid }}_{{ openshift_aws_node_group_type }}"  # node type sg
-      - "{{ openshift_aws_clusterid }}_{{ openshift_aws_node_group_type }}_k8s"  # node type sg k8s
+      group-name: "{{ openshift_aws_launch_config_security_groups }}"
+      vpc-id: "{{ vpcout.vpcs[0].id }}"
```
Comment (michaelgugino) [inline roles/openshift_aws/tasks/launch_config.yml]: This line is necessary because there can be duplicate security group names in different VPCs.
